src/ccl_haloprofile.c

```diff
@@ -7,349 +7,88 @@
-                          double massdef_delta_m, double a, double *r, int nr,
-                          double *rho_r, int *status) {
-
-    //haloradius: halo radius for mass definition
-    //rs: scale radius
-    double haloradius, rs;
-
-    haloradius = r_delta(cosmo, halomass, a, massdef_delta_m, status);
-    rs = haloradius/c;
-
-    //rhos: NFW density parameter
-    double rhos;
-
-    rhos = halomass/(4.*M_PI*(rs*rs*rs)*helper_fx(c));
-
-    int i;
-    double x;
-    for(i=0; i < nr; i++) {
-        x = r[i]/rs;
-        rho_r[i] = rhos/(x*(1.+x)*(1.+x));
-    }
-    return;
-}
-
-//cosmo: ccl cosmology object containing cosmological parameters
-//c: halo concentration, needs to be consistent with halo size definition
-//halomass: halo mass
-//massdef_delta: mass definition, overdensity relative to mean matter density
-//a: scale factor
-//rp: radius at which to calculate output
-//nr: number of radii for calculation
-//sigma_r: stores surface mass density (integrated along line of sight) at given projected rp
-//returns void
-void ccl_projected_halo_profile_nfw(ccl_cosmology *cosmo, double c,
-                                    double halomass, double massdef_delta_m,
-                                    double a, double *rp, int nr, double *sigma_r,
-                                    int *status) {
-
-    //haloradius: halo radius for mass definition
-    //rs: scale radius
-    double haloradius, rs;
-
-    haloradius = r_delta(cosmo, halomass, a, massdef_delta_m, status);
-    rs = haloradius/c;
-
-    //rhos: NFW density parameter
-    double rhos;
-
-    rhos = halomass/(4.*M_PI*(rs*rs*rs)*helper_fx(c));
-
-    double x;
-
-    int i;
-    for(i=0; i < nr; i++){
-
-        x = rp[i]/rs;
-        if (x==1.){
-            sigma_r[i] = 2.*rs*rhos/3.;
-        }
-        else if (x<1.){
-            sigma_r[i] = 2.*rs*rhos*(1.-2.*atanh(sqrt(fabs((1.-x)/(1.+x))))/sqrt(fabs(1.-x*x)))/(x*x-1.);
-        }
-        else {
-            sigma_r[i] = 2.*rs*rhos*(1.-2.*atan(sqrt(fabs((1.-x)/(1.+x))))/sqrt(fabs(1.-x*x)))/(x*x-1.);
-        }
-    }
-    return;
-
-}
-
-//maths helper function assuming NFW approximation
-static double helper_solve_cvir(double c, void *rhs_pointer){
-    double rhs = *(double*)rhs_pointer;
-    return (log(1.+c)-c/(1.+c))/(c*c*c) - rhs;
-}
-
-//solve for cvir from different mass definition iteratively, assuming NFW profile.
-static double solve_cvir(ccl_cosmology *cosmo, double rhs, double initial_guess, int *status){
-
-    int rootstatus;
-    int iter = 0, max_iter = 100;
-    const gsl_root_fsolver_type *T;
-    gsl_root_fsolver *s;
-    double cvir;
-    double c_lo = 0.1, c_hi = initial_guess*100.;
+void ccl_einasto_norm_integral(int n_m, double *r_s, double *r_delta, double *alpha,
+			       double *norm_out,int *status)
+{
+#pragma omp parallel default(none)			\
+  shared(n_m, r_s, r_delta, alpha, norm_out, status)
+  {
+    int ii;
```

Comment:
Yes. I prefer to do things this way (and I've seen this being encouraged), since it forces me to think about which variables should be shared and which ones shouldn't. `default(none)` forces you to declare absolutely all variables as shared or private, and anything inside the brackets is private.

Reply:
Got it. I always use `default(none)` but have been in the habit of declaring stuff outside the parallel region. Pro tip! Thanks!